test/constraint_violation.jl

```diff
@@ -0,0 +1,65 @@
+using MultilinearOpt
+using JuMP
+using Gurobi
+using Test
+using Random
+using LinearAlgebra
+
+function test_constraint_violation(f, xbounds::NTuple{2}, ybounds::NTuple{2};
+        method::Symbol, disc_level::Integer, rng::AbstractRNG, atol::Real, num_tests::Integer)
```

Comment:
The implicit assumption here is that that ``rng`` returns values in [0,1], correct?

Reply:
That `rand` does, yes. The documentation of `rand` specifies that the values are in [0, 1).